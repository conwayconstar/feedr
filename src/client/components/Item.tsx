import React from 'react';
import { Item as ItemType } from 'feedr';

export type ItemProps = { item: ItemType, menu?: boolean, toggleItem: (item: ItemType) => void; };

const Item: React.FC<ItemProps> = ({
  item, menu, toggleItem,
}) => {
  const handleClick = () => toggleItem(item);

  return (
    <>
      <li className="item" onClick={!menu ? handleClick : undefined}>
        <h2>{item.name}</h2>
        <p>
          {item.dietaries.map(((dietary) => (
            <span className="dietary" key={dietary}>{dietary}</span>
          )))}
        </p>
        {menu && (
        <button
          type="button"
          className="remove-item"
          onClick={handleClick}
        >
          x
        </button>
        )}
      </li>
    </>
  );
};

export default Item;
