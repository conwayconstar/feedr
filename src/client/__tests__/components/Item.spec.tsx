import { shallow } from 'enzyme';
import React from 'react';
import { Item as ItemType } from 'feedr';
import Item, { ItemProps } from '../../components/Item';

describe('Item component', () => {
  const toggleItem = jest.fn((item) => item);
  const item: ItemType = {
    id: 1001,
    name: 'Kale Caesar Pasta, Turmeric Satay Broccoli & Lemon Cashew Greens',
    dietaries: ['v', 've', 'df', 'gf', 'n!'],
  };
  const props: ItemProps = {
    item,
    toggleItem,
  };
  const wrapper = shallow(<Item {...props} />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Item should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('toggleItem should be called', () => {
    const itemEl = wrapper.find('.item');
    itemEl.simulate('click');
    expect(toggleItem.mock.results[0].value).toEqual(item);
  });

  it('Item component inside the menu', () => {
    wrapper.setProps({ menu: true });
    wrapper.update();
    const button = wrapper.find('.remove-item');
    expect(wrapper.find('.item').prop('onClick')).toEqual(undefined);
    expect(button.length).toEqual(1);
    button.simulate('click');
    expect(toggleItem.mock.results[0].value).toEqual(item);
  });
});
