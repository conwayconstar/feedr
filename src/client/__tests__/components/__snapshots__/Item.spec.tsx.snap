// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component Item should match snapshot 1`] = `
<Fragment>
  <li
    className="item"
    onClick={[Function]}
  >
    <h2>
      Kale Caesar Pasta, Turmeric Satay Broccoli & Lemon Cashew Greens
    </h2>
    <p>
      <span
        className="dietary"
        key="v"
      >
        v
      </span>
      <span
        className="dietary"
        key="ve"
      >
        ve
      </span>
      <span
        className="dietary"
        key="df"
      >
        df
      </span>
      <span
        className="dietary"
        key="gf"
      >
        gf
      </span>
      <span
        className="dietary"
        key="n!"
      >
        n!
      </span>
    </p>
  </li>
</Fragment>
`;
